src/EnergyPlus/HVACSizingSimulationManager.cc

```diff
@@ -0,0 +1,406 @@
+					strip(DayOfSimChr);
+					if (!WarmupFlag) {
+						++CurrentOverallSimDay;
+						DisplaySimDaysProgress(CurrentOverallSimDay, TotalOverallSimDays);
+					}
+					else {
+						DayOfSimChr = "0";
+					}
+					BeginDayFlag = true;
+					EndDayFlag = false;
+
+					if (WarmupFlag) {
+						++NumOfWarmupDays;
+						cWarmupDay = TrimSigDigits(NumOfWarmupDays);
+						DisplayString("Warming up {" + cWarmupDay + '}');
+					}
+					else if (DayOfSim == 1) {
+						DisplayString("Starting HVAC Sizing Simulation at " + CurMnDy + " for " + EnvironmentName);
+						gio::write(OutputFileInits, Format_700) << NumOfWarmupDays;
+					}
+					else if (DisplayPerfSimulationFlag) {
+						DisplayString("Continuing Simulation at " + CurMnDy + " for " + EnvironmentName);
+						DisplayPerfSimulationFlag = false;
+					}
+
+					for (HourOfDay = 1; HourOfDay <= 24; ++HourOfDay) { // Begin hour loop ...
+
+						BeginHourFlag = true;
+						EndHourFlag = false;
+
+						for (TimeStep = 1; TimeStep <= NumOfTimeStepInHour; ++TimeStep) {
+							if (AnySlabsInModel || AnyBasementsInModel){
+								InitAndSimGroundDomains();
+							}
+
+							BeginTimeStepFlag = true;
+
+
+							// Set the End__Flag variables to true if necessary.  Note that
+							// each flag builds on the previous level.  EndDayFlag cannot be
+							// .TRUE. unless EndHourFlag is also .TRUE., etc.  Note that the
+							// EndEnvrnFlag and the EndSimFlag cannot be set during warmup.
+							// Note also that BeginTimeStepFlag, EndTimeStepFlag, and the
+							// SubTimeStepFlags can/will be set/reset in the HVAC Manager.
+
+							if (TimeStep == NumOfTimeStepInHour) {
+								EndHourFlag = true;
+								if (HourOfDay == 24) {
+									EndDayFlag = true;
+									if ((!WarmupFlag) && (DayOfSim == NumOfDayInEnvrn)) {
+										EndEnvrnFlag = true;
+									}
+								}
+							}
+
+							ManageWeather();
+
+							ManageExteriorEnergyUse();
+
+							ManageHeatBalance();
+
+							BeginHourFlag = false;
+							BeginDayFlag = false;
+							BeginEnvrnFlag = false;
+							BeginSimFlag = false;
+							BeginFullSimFlag = false;
+
+						} // TimeStep loop
+
+						PreviousHour = HourOfDay;
+
+					} // ... End hour loop.
+					if (ReportDuringHVACSizingSimulation) {
+						if ( sqlite ) sqlite->sqliteCommit(); // one transaction per day
+					}
+				} // ... End day loop.
+
+
+			} // ... End environment loop.
+
+			sizeSimManagerObj.PostProcessLogs();
+
+			sizeSimManagerObj.ProcessCoincidentPlantSizeAdjustments( HVACSizingIterCount );
+
+			sizeSimManagerObj.RedoKickOffAndResize();
+
+			if ( ! sizeSimManagerObj.plantCoinAnalyRequestsAnotherIteration ) {
+				// jump out of for loop, or change for to a while
+				break;
+			}
+
+			sizeSimManagerObj.sizingLogger.IncrementSizingPeriodSet ( );
+
+		} // End HVAC Sizing Iteration loop
+
+
+		WarmupFlag = false;
+		DoOutputReporting = true;
+		sizeSimManagerObj.~HVACSizingSimulationManager(); //call destructor to free memory etc. 
```

Comment:
Ohh I see what you mean since you declared sizeSimManagerObj in global EnergyPlus::HVACSizingSimulationManagerNamespace namespace. After you are finished with hvac sizing you want to free that memory since E+ continues on with its simulations. Hummm... ok. Let me think about this since this is still an odd paradigm.